test/dependency_source/test_requirement.py

```diff
@@ -170,7 +170,7 @@ def _check_fixes(
     fixes: list[ResolvedFixVersion],
 ) -> None:
     # Populate the requirements files
-    for input_req, req_path in zip(input_reqs, req_paths):
+    for input_req, req_path in zip(input_reqs, req_paths, strict=False):
```

Comment:
```suggestion
    for input_req, req_path in zip(input_reqs, req_paths, strict=True):
```

(and the other callsites below)